### PR TITLE
Update to Python 3.8 minimum.

### PR DESCRIPTION
**Pull request by bjrichardwebster**

### CODE REVIEW

**Comment by Purg** [comment]
So I realized that an updated YAML config for CI will get picked up if the branch is pushed to the repository. I think that would be pertinent in this case and you should now have permission to do that now.

File: pyproject.toml [hunk 1]
```diff
@@ -18,39 +18,43 @@ classifiers = [
...
 [tool.poetry.dependencies]
-python = "^3.6"
-numpy = "^1.19.5"
-pillow = "^8.3.2"
-smqtk-core = ">=0.18.0"
-smqtk-dataprovider = ">=0.16.0"
+python = "^3.8"
+pillow = ">=9.2.0"
+numpy = ">=1.22"
+smqtk-core = "*"
```
Comment (Purg) [inline pyproject.toml:31]: I'm not sure my feelings on `"*"` usage have fully annealed, but I guess I'm okay with it as long as we are not aware of, or able to isolate, a minimum feature requirement. In this case, I think we want to base ourselves on `>=0.19` of smqtk-core as there is a behavior change in the Pluggable that removed some wonky `__new__` overriding.

File: pyproject.toml [hunk 2]
```diff
@@ -18,39 +18,43 @@ classifiers = [
...
 
 
 [tool.poetry.dev-dependencies]
 # CI
-flake8 = "^3.9.0"
-flake8-mutable = "^1.2.0"
-mypy = "^0.812"
+flake8 = "*"
+flake8-mutable = "*"
+mypy = "*"
```
Comment (Purg) [inline pyproject.toml:39]: For things like this I think I'm much more comfortable with the `"*"` declaration as they are more utilities that we use as opposed to software. The pinned versions in the lock file are in effect solely for reproducibility of results between developers and CI.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -52,22 +52,29 @@ test:build_docs:
     - make html
     - stat _build/html/index.html
 
-test:py36:
+test:py37:
```
Comment (Purg) [inline .gitlab-ci.yml:55]: It looks like the test job names are not aligned with the image being used (off-by-one too low).
```suggestion
test:py38:
```

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -52,22 +52,29 @@ test:build_docs:
...
   extends: .test_job_defaults
   script:
     - *test_setup  # expand `&test_setup` anchor above into here.
     - pytest --cov=smqtk --cov-config=.pytest.coveragerc
 
-test:py37:
+test:py38:
```
Comment (Purg) [inline .gitlab-ci.yml:61]:
```suggestion
test:py39:
```

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -52,22 +52,29 @@ test:build_docs:
...
   extends: .test_job_defaults
-  image: python:3.7-buster
+  image: python:3.9-buster
   script:
     - *test_setup  # expand `&test_setup` anchor above into here.
     - pytest --cov=smqtk --cov-config=.pytest.coveragerc
 
-test:py38:
+test:py39:
```
Comment (Purg) [inline .gitlab-ci.yml:68]:
```suggestion
test:py310:
```

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -52,22 +52,29 @@ test:build_docs:
...
   extends: .test_job_defaults
-  image: python:3.8-buster
+  image: python:3.10-buster
+  script:
+    - *test_setup  # expand `&test_setup` anchor above into here.
+    - pytest --cov=smqtk --cov-config=.pytest.coveragerc
+
+test:py310:
```
Comment (Purg) [inline .gitlab-ci.yml:75]:
```suggestion
test:py311:
```

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -4,7 +4,7 @@ stages:
 # Global default parameters set for all jobs unless overridden by job-specific
 # configuration.
 default:
-  image: python:3.6-buster
+  image: python:3.8-buster
```
Comment (Purg) [inline .gitlab-ci.yml:7]: I don't think the use of the `...-buster` naming here is necessary. I'm not sure why I did it in the first place and the image name `python:3.8` is wholly valid.

**Comment by bjrichardwebster** [comment]
Closed in lieu of the direct push to origin with #30 .

**Pull request closed without merging**